src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/OverlayPane.java

```diff
@@ -42,11 +42,15 @@ private Area getDimmedArea() {
     var overlayArea = new Area(new Rectangle(0, 0, getWidth(), getHeight()));
 
     for (var c : highlighters) {
-      // convertPoint is necessary because the component uses a different coordinate origin
+      var posDiff = SwingUtilities.convertPoint(c.getComponent(), 0, 0, this);
+      var translation = AffineTransform.getTranslateInstance(posDiff.x, posDiff.y);
+
       if (c.getComponent().isShowing()) {
         for (var rectangle : c.getArea()) {
-          var windowRect = SwingUtilities.convertRectangle(c.getComponent(), rectangle, this);
-          overlayArea.subtract(new Area(windowRect));
+          var rectangleArea = new Area(rectangle);
+          rectangleArea.transform(translation);
+
+          overlayArea.subtract(rectangleArea);
         }
       }
```

Comment:
This is just optimizing, right? No functional changes here? But good optimizing! :+1:

Reply:
`convertRectangle` only works for Rectangles. `AffineTransform` works on all RectangularShapes, including RoundedRectangle2D. Jaakko wanted rounded rectangles at some point, so this is a necessary change.

It also allows us to highlight buttons with more precision (since buttons are rounded rectangles in IntelliJ).